--- 23.0.0/src/amdgpu_kms.c	2023-02-22 10:07:05.000000000 -0700
+++ amdgpu_kms-patched.c	2025-04-13 14:46:02.071529943 -0600
@@ -1510,14 +1510,14 @@
 	else if (info->family < AMDGPU_FAMILY_CI)
 		info->cursor_w = CURSOR_WIDTH;
 	else
-		info->cursor_w = CURSOR_WIDTH_CIK;
+		info->cursor_w = CURSOR_WIDTH;
 
 	if (drmGetCap(fd, DRM_CAP_CURSOR_HEIGHT, &value) == 0)
 		info->cursor_h = value;
 	else if (info->family < AMDGPU_FAMILY_CI)
 		info->cursor_h = CURSOR_HEIGHT;
 	else
-		info->cursor_h = CURSOR_HEIGHT_CIK;
+		info->cursor_h = CURSOR_HEIGHT;
 }
 
 /* When the root window is mapped, set the initial modes */
