--- mesa/include/pci_ids/radeonsi_pci_ids.h	2025-04-02 10:35:11.000000000 -0600
+++ radeonsi_pci_ids-patched.h	2025-04-12 20:32:25.584568630 -0600
@@ -88,6 +88,11 @@
 CHIPSET(0x665D, BONAIRE)
 CHIPSET(0x665F, BONAIRE)
 
+CHIPSET(0x9920, LIVERPOOL)
+CHIPSET(0x9922, LIVERPOOL)
+CHIPSET(0x9923, LIVERPOOL)
+CHIPSET(0x9924, GLADIUS)
+
 CHIPSET(0x9830, KABINI)
 CHIPSET(0x9831, KABINI)
 CHIPSET(0x9832, KABINI)
--- mesa/src/amd/addrlib/src/amdgpu_asic_addr.h	2025-04-02 10:35:11.000000000 -0600
+++ amdgpu_asic_addr-patched.h	2025-04-12 20:39:27.285639884 -0600
@@ -62,7 +62,9 @@
 #define AMDGPU_HAWAII_RANGE     0x28, 0x3C //# 40 <= x < 60
 
 #define AMDGPU_SPECTRE_RANGE    0x01, 0x41 //#   1 <= x < 65
-#define AMDGPU_SPOOKY_RANGE     0x41, 0x81 //#  65 <= x < 129
+#define AMDGPU_SPOOKY_RANGE     0x41, 0x61 //#  65 <= x < 97
+#define AMDGPU_STARSHA_RANGE    0x61, 0x71 //#  97 <= x < 113
+#define AMDGPU_STARSHP_RANGE    0x71, 0x81 //# 113 <= x < 129
 #define AMDGPU_KALINDI_RANGE    0x81, 0xA1 //# 129 <= x < 161
 #define AMDGPU_GODAVARI_RANGE   0xA1, 0xFF //# 161 <= x < max
 
@@ -134,6 +136,8 @@
 
 #define ASICREV_IS_SPECTRE(r)          ASICREV_IS(r, SPECTRE)
 #define ASICREV_IS_SPOOKY(r)           ASICREV_IS(r, SPOOKY)
+#define ASICREV_IS_STARSHA(r)          ASICREV_IS(r, STARSHA)
+#define ASICREV_IS_STARSHP(r)          ASICREV_IS(r, STARSHP)
 #define ASICREV_IS_KALINDI(r)          ASICREV_IS(r, KALINDI)
 #define ASICREV_IS_KALINDI_GODAVARI(r) ASICREV_IS(r, GODAVARI)
 
--- mesa/src/amd/common/ac_gpu_info.c	2025-04-02 10:35:11.000000000 -0600
+++ ac_gpu_info-patched.c	2025-04-12 20:46:27.543346927 -0600
@@ -771,6 +771,8 @@
       case FAMILY_KV:
          identify_chip2(SPECTRE, KAVERI);
          identify_chip2(SPOOKY, KAVERI);
+         identify_chip2(STARSHA, LIVERPOOL);
+         identify_chip2(STARSHP, GLADIUS);
          identify_chip2(KALINDI, KABINI);
          identify_chip2(GODAVARI, KABINI);
          break;
@@ -1111,6 +1113,8 @@
       case CHIP_VERDE:
       case CHIP_HAINAN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
+      case CHIP_GLADIUS:
       case CHIP_KAVERI:
       case CHIP_ICELAND:
       case CHIP_CARRIZO:
@@ -2165,6 +2169,8 @@
    case CHIP_PITCAIRN:
    case CHIP_VERDE:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_HAWAII:
    case CHIP_TONGA:
    case CHIP_FIJI:
@@ -2215,6 +2221,14 @@
       raster_config = 0x16000012;
       raster_config_1 = 0x00000000;
       break;
+   case CHIP_LIVERPOOL:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x00000000;
+      break;
+   case CHIP_GLADIUS:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x0000002e;
+      break;
    /* 2 SEs / 8 RBs */
    case CHIP_TAHITI:
    case CHIP_PITCAIRN:
--- mesa/src/amd/common/amd_family.h	2025-04-02 10:35:11.000000000 -0600
+++ amd_family-patched.h	2025-04-12 20:48:03.635761678 -0600
@@ -80,6 +80,8 @@
    CHIP_HAINAN,
    /* GFX7 (Sea Islands) */
    CHIP_BONAIRE,
+   CHIP_LIVERPOOL,
+   CHIP_GLADIUS,
    CHIP_KAVERI,
    CHIP_KABINI,
    CHIP_HAWAII,         /* Radeon 290, 390 */
--- mesa/src/amd/vulkan/winsys/null/radv_null_winsys.c	2025-04-02 10:35:11.000000000 -0600
+++ radv_null_winsys-patched.c	2025-04-12 21:02:40.474127355 -0600
@@ -27,6 +27,10 @@
    [CHIP_OLAND] = {0x6060, 2, true},
    [CHIP_HAINAN] = {0x6660, 2, true},
    [CHIP_BONAIRE] = {0x6640, 4, true},
+   [CHIP_LIVERPOOL] = { 0x9920, 2, false },
+   [CHIP_LIVERPOOL] = { 0x9922, 2, false },
+   [CHIP_LIVERPOOL] = { 0x9923, 2, false },
+   [CHIP_GLADIUS] = { 0x9924, 4, false },
    [CHIP_KAVERI] = {0x1304, 2, false},
    [CHIP_KABINI] = {0x9830, 2, false},
    [CHIP_HAWAII] = {0x67A0, 16, true},
--- mesa/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c	2025-04-02 10:35:11.000000000 -0600
+++ radeon_drm_winsys-patched.c	2025-04-12 21:08:07.733237475 -0600
@@ -259,6 +259,8 @@
       ws->info.gfx_level = GFX6;
       break;
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
    case CHIP_KABINI:
    case CHIP_HAWAII:
@@ -472,6 +474,8 @@
    case CHIP_VERDE:
    case CHIP_OLAND:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
       ws->info.max_tcc_blocks = 4;
       break;
@@ -501,9 +505,11 @@
       case CHIP_TAHITI:
       case CHIP_PITCAIRN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
          ws->info.max_se = 2;
          break;
       case CHIP_HAWAII:
+      case CHIP_GLADIUS:
          ws->info.max_se = 4;
          break;
       }
--- mesa/src/gallium/winsys/radeon/drm/radeon_surface.c	2025-04-12 23:14:06.779284863 -0600
+++ radeon_surface.c	2025-04-12 23:12:07.574346068 -0600
@@ -78,6 +78,8 @@
     CHIP_OLAND,
     CHIP_HAINAN,
     CHIP_BONAIRE,
+    CHIP_LIVERPOOL,
+    CHIP_GLADIUS,
     CHIP_KAVERI,
     CHIP_KABINI,
     CHIP_HAWAII,
--- mesa/src/amd/common/amd_family.c	2025-04-22 12:40:01.508061990 -0600
+++ amd_family.c	2025-04-22 14:59:28.582004242 -0600
@@ -19,6 +19,8 @@
    CASE(OLAND);
    CASE(HAINAN);
    CASE(BONAIRE);
+   CASE(LIVERPOOL);
+   CASE(GLADIUS);
    CASE(KABINI);
    CASE(KAVERI);
    CASE(HAWAII);
@@ -105,6 +107,10 @@
       return "hainan";
    case CHIP_BONAIRE:
       return "bonaire";
+   case CHIP_LIVERPOOL:
+      return "liverpool";
+   case CHIP_GLADIUS:
+      return "gladius";
    case CHIP_KABINI:
       return "kabini";
    case CHIP_KAVERI:
